pageserver/Cargo.toml

```diff
@@ -75,6 +75,7 @@ enum-map.workspace = true
 enumset.workspace = true
 strum.workspace = true
 strum_macros.workspace = true
+take_mut = "0.2.2"
```

Comment:
I think we covered that this is not needed, it also has the nice abort-y behaviour.

Reply:
Let's continue conversation in the other take_mut comment.